premium/offline.go

```diff
@@ -80,6 +80,10 @@ func WithAWSLicenseManagerClient(awsLicenseManagerClient AWSLicenseManagerInterf
 	}
 }
 
+func isMarketplaceLicense() bool {
```

Comment:
since os.Getenv wouldn't return different things, I would add this as a`bool` in `CQLicenseClient` and initialize it on the same line:
```go
cl := CQLicenseClient{
		logger: logger,
		isMarketplaceLicense: os.Getenv("CQ_AWS_MARKETPLACE_LICENSE") == "true",
...
}
```

Reply:
You could also keep this as a local `bool` (and init the aws client on that), and depend on the fact that `cl. awsLicenseManagerClient` being `nil` or not inside `ValidateLicense()`.